charms/worker/k8s/src/token_distributor.py

```diff
@@ -164,24 +164,24 @@ def _create_cos_token(self, name: str, _):
             username=f"system:cos:{name}", groups=["system:cos"]
         )
 
-    def _revoke_cluster_token(self, name: str, _):
+    def _revoke_cluster_token(self, name: str, ignore_errors: bool):
```

Comment:
OK, there may be other reasons we want to ignore errors other than just i'm the leader and i want out of the cluster.  For instance -- we can try to revoke the token of a node that half-joined, or failed to join.  This lets us effectively "try" to remove this node from the cluster.  More below